src/wayland/dmabuf/dispatch.rs

```diff
@@ -97,23 +182,27 @@ where
             formats: global_data.formats.clone(),
             id: global_data.id,
             logger: global_data.logger.clone(),
+            default_feedback: global_data.default_feedback.clone(),
+            known_default_feedbacks: global_data.known_default_feedbacks.clone(),
         };
 
         let zwp_dmabuf = data_init.init(resource, data);
 
         // Immediately send format info to the client if we are the correct version.
         //
         // These events are deprecated in version 4 of the protocol.
-        if zwp_dmabuf.version() <= 3 {
+        if zwp_dmabuf.version() < zwp_linux_dmabuf_v1::REQ_GET_DEFAULT_FEEDBACK_SINCE {
             for format in &*global_data.formats {
                 zwp_dmabuf.format(format.code as u32);
```

Comment:
I guess this results in formats that are supported with multiple modifiers being sent multiple times? Is it possible for a DRM format to only work with an explicit modifier? In which case this could also send an incorrect format.

Though this is existing code, not added by this PR.

Reply:
Right the filtering for dmabuf < v3 is not good here.

Reply:
should be fixed in latest version